### ISSUE TITLE
ManagedIdentity for VMSS is not shown as SystemAssigned

### ISSUE THREAD

**Issue by NileshGule** [Azure/secrets-store-csi-driver-provider-azure#135]
**What steps did you take and what happened:**
Created an AKS cluster with Managed Identity, Kubernetes versions 1.16.9. Below are the environment details. Is there anything else I need to do to assign managed Identity to the nodepool?

**What did you expect to happen:**
Tried to get the identity assigned to the nodepool as specified in the [link](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/docs/system-assigned-msi-mode.md). Expected the type of identity as SystemAssigned, but actual result is **UserAssigned**

**Anything else you would like to add:**
NA

**Which access mode did you use to access the Azure Key Vault instance:**
System Assigned Managed Identity
<img width="651" alt="AKS create with managed identity" src="https://user-images.githubusercontent.com/1298374/85548039-1ccca000-b651-11ea-800c-8f243ffef567.PNG">
<img width="1859" alt="AKV AKS managed identity error" src="https://user-images.githubusercontent.com/1298374/85548045-1dfdcd00-b651-11ea-8789-5a17dd170db6.PNG">

**Environment:**

- Secrets Store CSI Driver version: (use the image tag):
- Azure Key Vault provider version: (use the image tag):
- Kubernetes version: 1.18.2 and `kubectl get nodes -o wide`):
- Cluster type: AKS

<img width="1911" alt="kubectl output" src="https://user-images.githubusercontent.com/1298374/85548415-7fbe3700-b651-11ea-9250-3ff8f793c1c9.PNG">

**Comment by aramase** [comment]
@jluk Should we update the doc?

This is what it currently says, but that doesn't match the output @NileshGule posted. 

```
For AKS clusters, system assigned managed identities can be created on your behalf. To learn more read here about AKS managed identity support.
```

```
AKS does not support user assigned managed identities yet, only system assigned managed identities. Until this gap is covered, use a service principal or system assigned identities with AKS.
```

**Comment by jluk** [comment]
@tomgeske / @mlearned can you update the MSI doc based on this feedback, I agree the terminology is misleading. Perhaps clarifying custom BYO identities are what are not supported.

**Comment by TomGeske** [comment]
Yes, agree, we should align terminology in CSI driver docs with our MI docs https://aka.ms/aks/mi

**Comment by aramase** [comment]
> Perhaps clarifying custom BYO identities are what are not supported.

I think in the context of CSI driver this is not a limitation that the Azure Key Vault Provider doc needs to call out as the objective here s for the user to assign the MSI manually to the VMSS or use pod-identity to assign the MSI.

**Comment by NileshGule** [comment]
@aramase I don't think user need to manually assign the MSI to VMSS. When I create the cluster using `assign-managed-identities` option the VMSS pool is assigned an Identity. I granted the access to AKV using access policy to this identity and was able to pull the secrets at the time of starting the pod.

I demonstrated this in a YouTube video https://youtu.be/MHm4IVGVO1w

Let me know if my understanding is wrong.

**Comment by aramase** [comment]
> @aramase I don't think user need to manually assign the MSI to VMSS. When I create the cluster using enable-managed-identities option the VMSS pool is assigned an Identity. I granted the access to AKV using access policy to this identity and was able to pull the secrets at the time of starting the pod.


@NileshGule That's correct. Cluster with `enable-managed-identities` will contain a cluster User MSI. You can use the cluster MSI to access Key Vault contents or create another managed identity with scoped permissions only for Key Vault. If you create another MSI, then that needs to be manually assigned. 

We have that documented here - https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/docs/user-assigned-msi-mode.md#vmss-user-assigned-managed-identity

```
You can create AKS with managed identities now and then you can skip steps 1 and 2. To be able to get the CLIENT ID, the user can run the following command

az aks show -g <resource group> -n <aks cluster name> --query identityProfile.kubeletidentity.clientId -o tsv
```

But I think we need to modify this snippet in the doc to align with https://aka.ms/aks/mi  - 

```
AKS does not support user assigned managed identities yet, only system assigned managed identities. Until this gap is covered, use a service principal or system assigned identities with AKS.
```

**Comment by TomGeske** [comment]
I just made a docs PR.

**Comment by aramase** [comment]
Closed with #154
